src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,18 +32,27 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={event => {
+            dispatch(filterSlice.actions.setQuery(event.target.value));
```

Comment:
Ensure that the `setQuery` action is correctly defined in your `filterSlice`. This line updates the query state based on user input.